### ISSUE TITLE
MD038: False positive on code spans split across lines

### ISSUE THREAD

**Issue by joshrotenberg** [joshrotenberg/mdbook-lint#277]
## Summary

MD038 (no-space-in-code) incorrectly flags code spans that are split across lines in the source markdown.

## Example

```markdown
Using 'cargo run' is more convenient than having to remember to run 'cargo
build' and then use the whole path to the binary
```

We flag this as "Spaces inside code span elements" because we're detecting a code span from the backtick before "cargo" on line 1 to the backtick after "build" on line 2.

But these are actually TWO separate code spans:
- 'cargo run'
- 'cargo build'

The word "cargo" at end of line 1 and "build" at start of line 2 should not be parsed as a single code span.

## Impact

| Linter | MD038 Violations |
|--------|------------------|
| mdbook-lint | 37 |
| rumdl | 0 |
| markdownlint | 0 |

## Root Cause

Our code span parsing doesn't properly handle line breaks. A backtick followed by a newline before the closing backtick should not be treated as a valid code span (or the newline should not count as "space inside").

## Related

Issue #263 - Linter comparison

**Comment by joshrotenberg** [comment]
Fixed in PR #279.
